### PR TITLE
Move some tests in src/test/compile-fail -> src/test/ui

### PR DESCRIPTION
**Pull request by Centril**
This is a subset of https://github.com/rust-lang/rust/pull/62113.

Hopefully everything here sticks, but if it doesn't I'll remove tests until it does.

r? @petrochenkov

### CODE REVIEW

**Comment by Centril** [comment]
Yay! Initial tests seem to pass at least.

**Comment by petrochenkov** [comment]
Some of these are going to fail on wasm, IIRC.
If you are on Linux, could you run the docker image for the wasm target and see whether it fails?

**Comment by petrochenkov** [comment]
r=me after the wasm check

**Comment by Centril** [comment]
> If you are on Linux, could you run the docker image for the wasm target and see whether it fails?

@petrochenkov Will do. Should I ignore tests for the wasm target if they fail?

**Comment by petrochenkov** [comment]
Splitting into `ignore-wasm` and `only-wasm` would be better.

**Comment by petrochenkov** [comment]
(Since they "fail" by producing platform-dependent output, rather than by really failing.)

**Comment by Centril** [comment]
Tested wasm with `./src/ci/docker/run.sh test-various` and made some fixes. Hopefully this will stick now but I won't say it's a sure thing...

@bors r=petrochenkov

File: src/test/ui/imports/empty-extern-arg-wasm.stderr [hunk 1]
```diff
@@ -0,0 +1,9 @@
+error: extern location for std does not exist: 
+
+error[E0463]: can't find crate for `std`
+   |
+   = note: the `wasm32-unknown-unknown` target may not be installed
```
Comment (RalfJung) [inline src/test/ui/imports/empty-extern-arg-wasm.stderr]: You could also use normalization to get rid of these two lines. Then you don't need to duplicate tests.

Comment (RalfJung) [reply src/test/ui/imports/empty-extern-arg-wasm.stderr]: I have e.g. sued the following to normalize away entire lines in other tests:
```
// normalize-stderr-test "thread.*panicked.*Metadata module not compiled.*\n" -> ""
// normalize-stderr-test "note:.*RUST_BACKTRACE=1.*\n" -> ""
```

Comment (petrochenkov) [reply src/test/ui/imports/empty-extern-arg-wasm.stderr]: Hm, this is great news.
For some reason I thought the normalization works per-line.

Comment (petrochenkov) [reply src/test/ui/imports/empty-extern-arg-wasm.stderr]: Much simpler to fix platform-dependent tests this way.

Comment (RalfJung) [reply src/test/ui/imports/empty-extern-arg-wasm.stderr]: I think the matching of the normalization pattern *is* per-line -- but if you include the final `\n`, you can remove the entire line. And then that's handled correctly when comparing later.

Comment (Centril) [reply src/test/ui/imports/empty-extern-arg-wasm.stderr]: I'll have a look at this tomorrow; in the meantime feel free to r- or we can do this in a follow up PR if that's alright with y'all.

**Comment by Centril** [comment]
@bors r-

Failed in https://github.com/rust-lang/rust/pull/62294#issuecomment-507569342.

Will try the normalization strategy instead.

**Comment by chocol4te** [comment]
@Centril Ping from triage regarding the merge conflicts! :)

**Comment by JohnCSimon** [comment]
@Centril @petrochenkov Ping from triage, this is so close! :)

**Comment by gagan0723** [comment]
Third ping from triage, @Centril any updates on this?

**Note**: Thanks for the PR. This will be closed and marked as `S-inactive-closed` next week. Feel free to re-open when you have time.

Thanks

**Comment by Centril** [comment]
Closing for now until I have time to get back to it :)

**Pull request closed without merging**